src/App.tsx

```diff
@@ -8,27 +12,105 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
+import { normalize } from './helpers/normalize';
+
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setIsLoaded(true);
+      });
+  }, []);
+
+  const selectTodo = useCallback((todo: Todo) => {
```

Comment:
probably I think we can remove all `useCallback` and `useMemo` in this application.

look if we change input or filter we re-render `TodoFilter` and `TodoList`. In case we open `TodoModal` we won't render TodoFilter but this component doesn't have hard logic so don't worry.